most-used-words.el

```diff
@@ -50,97 +50,170 @@
+
+;;; Tabluldated List
+
+(defconst most-used-words--format
+  (vector (list "Word" 15 t)
+          (list "Counts" 10 t)
+          (list "Percentages" 10 t))
+  "Format to assign to `tabulated-list-format' variable.")
+
+(defun most-used-words--get-entries ()
+  "Data entries for most used words."
+  (let ((most-used (most-used-words-data most-used-words-word-display))
+        (entries '()) (id-count 0))
+    (dolist (word-data most-used)
+      (let ((new-entry '()) (new-entry-value '())
+            (word (plist-get word-data :word))
+            (count (plist-get word-data :count))
+            (percent (plist-get word-data :percent)))
+        (push (concat (format "%0.2f" percent) "%") new-entry-value)  ; Percentages
+        (push (format "%s" count) new-entry-value)  ; Counts
+        (push word new-entry-value)  ; Word
+        (push (vconcat new-entry-value) new-entry)  ; Turn into vector.
+        (push (number-to-string id-count) new-entry)  ; ID
+        (push new-entry entries)
+        (setq id-count (1+ id-count))))
+    entries))
+
+(define-derived-mode most-used-words-mode tabulated-list-mode
+  "most-used-words-mode"
+  "Major mode for Most Used Words display."
+  :group 'most-used-words
+  (setq tabulated-list-format most-used-words--format)
+  (setq tabulated-list-padding 1)
+  (setq tabulated-list-sort-key (cons "Counts" nil))
+  (tabulated-list-init-header)
+  (setq tabulated-list-entries (most-used-words--get-entries))
+  (tabulated-list-print t))
+
+;;; Entry
 
 ;;;###autoload
 (defun most-used-words-buffer (&optional n)
   "Show the N (default 3) most used words in the current buffer."
   (interactive (list (completing-read
-		      "How many words? (default 3) "
-		      nil
-		      nil
-		      nil
-		      nil
-		      nil
-		      3
-		      nil)))
-  (unless (numberp n)
-    (setf n (string-to-number n)))
+                      (format "How many words? (default %s) "
+                              most-used-words-word-display)
+                      nil nil nil nil nil
+                      most-used-words-word-display
+                      nil)))
+  (unless (numberp n) (setf n (string-to-number n)))
   (most-used-words-buffer-aux n nil))
 
 ;;;###autoload
 (defun most-used-words-buffer-with-counts (&optional n)
-    "Show with counts the N (default 3) most used words in the current buffer."
+  "Show with counts the N (default 3) most used words in the current buffer."
   (interactive (list (completing-read
-		      "How many words? (default 3) "
-		      nil
-		      nil
-		      nil
-		      nil
-		      nil
-		      3
-		      nil)))
-  (unless (numberp n)
-    (setf n (string-to-number n)))
+                      (format "How many words? (default %s) "
+                              most-used-words-word-display)
+                      nil nil nil nil nil
+                      most-used-words-word-display
+                      nil)))
+  (unless (numberp n) (setf n (string-to-number n)))
```

Comment:
Use default display word! Easier for user to use! 😄

Reply:
Great.